hammer_cli_reference/playbook.yaml

```
@@ -0,0 +1,100 @@
+---
+- name: "Check prerequisites on developer-system"
+  connection: local
+  hosts: developer-system
+  vars:
+    repo_path: "/home/user"
+    repo_remote: "git@github.com:theforeman/foreman-documentation.git"
+
+  tasks:
+    - name: "Verify that user input is overwritten"
+      ansible.builtin.assert:
+        that: repo_path != "/home/user"
+
+    - name: "Verify that shellscript is present"
+      ansible.builtin.stat:
+        path: "{{ repo_path }}/guides/scripts/generate-hammer-reference.sh"
+      register: result
+
+    - name: "End playbook if shellscript is not present"
+      ansible.builtin.fail:
+        msg: "Shellscript to convert markdown to asciidoc is not present or executable"
+      when:
+        - not result.stat.executable
+        - not result.stat.exists
+
+    - name: "Delete markdown file if present"
+      ansible.builtin.file:
+        path: "{{ repo_path }}/guides/hammer_full_help.md"
+        state: absent
+
+    - name: "Verify that foreman-documentation is present"
+      ansible.builtin.git:
+        clone: false
+        dest: "{{ repo_path }}"
+        repo: "{{ repo_remote }}"
+        update: false  # noqa: latest
+        version: "HEAD"
+
+- name: "Generate Hammer CLI help"
+  hosts: foreman-katello
+
+  tasks:
+    - name: "Update DNF cache"
+      ansible.builtin.dnf:
+        update_cache: true
+
+    - name: "Install all available Hammer CLI packages"
+      ansible.builtin.package:
+        name: "rubygem-hammer_cli*"
+        state: latest  # noqa: package-latest
+
+    - name: "Generate full Hammer CLI help"
+      ansible.builtin.command:
+        cmd: "hammer full-help --md"
+      changed_when: true
+      register: hammer_cli_help_in_markdown
+
+    - name: "Write Hammer CLI help to file"
+      ansible.builtin.copy:
+        content: "{{ hammer_cli_help_in_markdown.stdout }}"
+        dest: "/tmp/hammer_full_help.md"
+        mode: "0644"
+
+    - name: "Copy markdown from orcharhino to local host"
+      ansible.posix.synchronize:
+        dest: "{{ repo_path }}/guides/hammer_full_help.md"
+        mode: pull
+        src: "/tmp/hammer_full_help.md"
```

```suggestion
      ansible.builtin.fetch:
        dest: "{{ repo_path }}/guides/hammer_full_help.md"
        src: "/tmp/hammer_full_help.md"
        flat: true
```

I prefer fetch over synchronize. Synchronize is a wrapper around rsync and requires it to be installed on the Ansible control node. `fetch` reduces the dependencies.